### PR TITLE
Add CPU option. Rip out deprecated apiVersion

### PR DESCRIPTION
**Pull request by inlined**
Also fixes a couple of build errors I was getting in identity.ts

### CODE REVIEW

File: integration_test/functions/src/v1/testLab-utils.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import fetch from 'node-fetch';
 import * as admin from 'firebase-admin';
+import fetch from 'node-fetch';
```
Comment (taeold) [inline integration_test/functions/src/v1/testLab-utils.ts:2]: q: do you have some eslint locally that automatically formats this? (or is it something you did manually)

File: src/runtime/loader.ts [hunk 2]
```diff
@@ -64,19 +64,21 @@ export function extractStack(
...
 ) {
-  for (const [name, val] of Object.entries(module)) {
+  for (const [name, valAsUnknown] of Object.entries(module)) {
+    // We're introspecting untrusted code here. Any is appropraite
+    const val: any = valAsUnknown;
     if (
       typeof val === 'function' &&
-      val['__endpoint'] &&
-      typeof val['__endpoint'] === 'object'
+      (val as any).__endpoint &&
```
Comment (taeold) [inline src/runtime/loader.ts:72]: nit* is `(val as any)` required here? I'd guess it's already any based on L69.

File: src/v2/options.ts [hunk 3]
```diff
@@ -167,9 +167,21 @@ export interface GlobalOptions {
...
    * Can only be applied to functions running on Cloud Functions v2.
-   * A value of null restores the default concurrency.
+   * A value of null restores the default concurrency (80 when CPU >= 1, 1 otherwise).
+   * Concurrency cannot be set to any value other than 1 if `cpu` is less than 1.
    */
   concurrency?: number | null;
+
+  /**
+   * Fractional number of CPUs to allocate to a function.
+   * defaults to 1 for functions with <= 2GB RAM and increases for larger memory sizes.
```
Comment (taeold) [inline src/v2/options.ts:177]:
```suggestion
   * Defaults to 1 for functions with <= 2GB RAM and increases for larger memory sizes.
```

**Review by taeold — APPROVED** [review]

File: integration_test/functions/src/v1/testLab-utils.ts (hunk 1 above)
Comment (inlined) [reply integration_test/functions/src/v1/testLab-utils.ts:2]: npm run format:fix && npm run lint:fix

File: src/runtime/loader.ts (hunk 2 above)
Comment (inlined) [reply src/runtime/loader.ts:72]: You're right. I was doing this manually and then used the any cast above. Will remove.

**Pull request merged**